src/System.Text.Primitives/System/Text/InvariantParser_signed.tt

```diff
@@ -0,0 +1,321 @@
+                        }
+                    }
+                    else if (value > <#= classes[i] #>.MaxValue / 10) // overflow
+                    {
+                        value = 0;
+                        bytesConsumed = 0;
+                        return false;
+                    }
+                    else if (value > 0 && (<#= classes[i] #>.MaxValue - value * 10) + (negative ? 1 : 0) < nextByteVal) // overflow (+1 allows for overflow for min value)
+                    {
+                        value = 0;
+                        bytesConsumed = 0;
+                        return false;
+                    }
+                    <#= primitives[i] #> candidate = (<#= primitives[i] #>)(value * 10 + nextByteVal); // parse the current digit to a <#= primitives[i] #> and add it to the left-shifted value
+                    value = candidate;
+                    bytesConsumed++; // increment the number of bytes consumed, then loop
+                }
+
+                if (negative && value != <#= classes[i] #>.MinValue) // We check if the value is negative at the very end to save on comp time
+                {
+                    value = (<#= primitives[i] #>)-value;
+                }
+                return true;
+            }
+            else if (cultureAndEncodingInfo.IsInvariantUtf16)
+            {
+                if (utf8Text[index] == '-')
+                {
+                    negative = true;
+                    signed = true;
+                    index += 2;
+                    bytesConsumed += 2;
+                }
+                else if (utf8Text[index] == '+')
+                {
+                    signed = true;
+                    index += 2;
+                    bytesConsumed += 2;
+                }
+
+                for (int byteIndex = index; byteIndex < utf8Text.Length - 1; byteIndex += 2) // loop through the byte array two bytes at a time for UTF-16
+                {
+                    byte byteAfterNext = utf8Text[byteIndex + 1];
+                    byte nextByteVal = (byte)(utf8Text[byteIndex] - '0');
+                    if (nextByteVal > 9 || byteAfterNext != 0) // if the second byte isn't zero, this isn't an ASCII-equivalent code unit and we can quit here
+															   // if nextByteVal > 9, we know it is not a digit because any value less than '0' will overflow
+															   // to greater than 9 since byte is an unsigned type.
+                    {
+                        if (bytesConsumed == 2 && signed) // if the first character happened to be a '-' or a '+', we reset the byte counter so logic proceeds as normal
+                        {
+                            bytesConsumed = 0;
+                        }
+                        if (bytesConsumed == 0) // check to see if we've processed any digits at all
+                        {
+                            value = default(int); // if we haven't, set value to 0 and return false
```

Comment:
>value = default(int); [](start = 28, length = 21)

this already initialized to zero in the beginning of  method. you don't need to reset i there.